### PR TITLE
refactor: create models package

### PR DESCRIPTION
**Pull request by felipehuerta17**
To facilitate testing and development, the modules
model.py and post_regression.py were moved to the
models folder.
    
An empty __init__.py was created in that folder

The Jupyter notebook was updated

### CODE REVIEW

**Comment by jia200x** [comment]
lint test failed

**Comment by jia200x** [comment]
the style issues are not related to this PR. However, we can "piggyback" this into a single commit of this PR? (e.g style(model): fix post_regression issues"

**Comment by jia200x** [comment]
Otherwise the proposed change makes sense and looks good! Thanks

**Comment by leandrolanzieri** [comment]
Would it make sense to split the models into files (e.g. `sir.py`, `sirx.py`) so we can export them at module level, and then use them
```python
from models import SIR
```
?

**Comment by jia200x** [comment]
> Would it make sense to split the models into files (e.g. sir.py, sirx.py) so we can export them at module level, and then use them

You are right, we should be able to import the modules with those import lines.

This pattern can be done with the __init__.py file. See e.g https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/covariance/__init__.py

**Comment by jia200x** [comment]
A good example on how to do this pattern is available at https://timothybramlett.com/How_to_create_a_Python_Package_with___init__py.html. Please refer to the "Adding import to __init__.py" section

**Comment by felipehuerta17** [comment]
@jia200x linting corrected
@leandrolanzieri sir and sirx separated.

I implemented slight changes in the API in how to call confidence intervals to use less parameters. Now, a dictionary options can be provided to the confidence intervals routines. If no dictionary is provided, the default parameters are used. @jia200x I will open soon another PR to continue on connecting the confidence intervals as model member functions.

**Comment by jia200x** [comment]
> I implemented slight changes in the API in how to call confidence intervals to use less parameters. 

Just a minor comment: an API change in a function is out of the scope of the PR. This PR should only "create models package".
I'm OK with piggybacking that commit, but for the next time I suggest to branch from this PR, open a second PR and add this as a dependency. See for instance https://github.com/open-sir/open-sir/pull/8, where I did this pattern.

For instance
```
# From `modules` branch
git checkout -b models_api_change
# Do some changes
git commit "feat(model): add support for dicts"
# Push and open another PR
git push origin models_api_change
# Go back to `models`
git checkout models
# Add more commits to models
git commit  "blah blah"
# Push again to models
git push origin models

# `models` get merged!!

# Update master
git checkout master
git pull origin master

# And now, rebase `models_api_change` with the new master
git rebase master

# Commit whatever is needed and then push with --force
git commit "asdf"
git push origin models_api_change --force
```

> @leandrolanzieri sir and sirx separated.

The current PR doesn't reflect the behavior that @leandrolanzieri describes. Basically we should be able to import like this:
```python
from models import SIR
```
instead of
```python
from models.sir import SIR
```

Otherwise, "sir" is redundant.

Basically doing `import models` calls the `__init__.py` file first, so you could do something like

__init__.py
```python
from sir import SIR
from sirx import SIRX
from model import ci_bootstrap, etc etc
```

And this would do the trick. Let me know if something is not clear :)

**Comment by felipehuerta17** [comment]
@jia200x although __init__.py already was compliant with your observation for the models, it wasn't for post_regression routines and now is updated. Many thanks for the thorough explanation.

Will do what you suggest on separation of concerns in PR

**Comment by jia200x** [comment]
This looks good. However, the number of commits should be reduced to the actual number of semantic units.

For instance, I see 2 main tasks
- Creating the package
- Adapting the Jupyter notebook.

(There are some style fixes, but let's just piggyback them into the "Creating the package" commit)
All the rest of the commits are just "fixups". So, the strategy is to squash all the fixups in the commit they belong.

Since commits are "file atomic" between the Jupyter and the rest (commit 1 and commit 2 are independent), there's a very easy trick to convert all commits in just 2 commits.

The idea is to reset the branch to the common ancestor between master and your branch (a.k.a the commit before your first commit of this PR).
Either you manually check it with `git log` or you use `git merge-base BRANCH_A BRANCH_B` to find the common ancestor between BRANCH_A and BRANCH_B.
For this case, BRANCH_A is your latest commit and BRANCH_B is master:
```
 git merge-base master HEAD
aa255e59ee20d2b69373f978eef7d6107cda3e66
```

Now, you reset to that commit (without --hard, otherwise you also drop the file changes!)

```
git reset aa255e59ee20d2b69373f978eef7d6107cda3e66
```
(tip: There's a cool one liner for this: `git reset $(git merge-base HEAD master)`)

The `git status` command shows:
```
	modified:   SIR-X.ipynb
	deleted:    model.py
	modified:   open-sir.py
	deleted:    post_regression.py

  (use "git add <file>..." to include in what will be committed)
	models/
```
Now the "models" folder is unstaged. We have to manually add it when we commit.

So now, it's possible to create both commits:
```
git add model.py open-sir.py post_regression.py models
git commit -m "refactor(model): create models package"
git add SIR-X.ipynb
git commit -m "refactor(jupyter): adapt to models package"
```
Done! `git log` shows:
```
c6e5345 (HEAD -> modules) refactor(jupyter): adapt to models package
00d9cfe refactor(model): create models package
aa255e5 (origin/master, origin/HEAD, master) feat(model): implement confidence intervals (#17)
```

Now, you need to force push, because the history in "upstream" completely diverges now.
Simply do:
```
git push origin modules --force
```

And that's all folks!

**Comment by felipehuerta17** [comment]
Hey @jia200x, everything very clear with the reset trick! I did it and it worked well. Also I understood it for future scenarios. Good to now the merge-base command!

I imagine this was necessary as otherwise git would automatically merge and squash two different things.

**Comment by jia200x** [comment]
Good to hear!

> I imagine this was necessary as otherwise git would automatically merge and squash two different things.

The commit history should reflect the semantic changes, so having to many commits make things harder when there's a bug introduced somewhere. That's why it's a good practice to keep the PRs with just a few commits.
Sometimes it's unavoidable (e.g changes that touch too many independent components), but in most cases this doesn't happen.

I suggest for the next time to have in mind the semantic commits, commit them first and then only add fixups.

For instance:
```
git commit --fixup HASH_FROM_COMMIT_TO_FIXUP
``` 

Then, the commit message reflects which commit belongs to.
